### DISCUSSION TITLE
Vacuum Polarization

### DISCUSSION THREAD

**Discussion by marti2216** [FeynCalc/feyncalc#115] [General]
Hello,
I have problems in computing the UV renormalization of the vacuum polarization. I find an explicit expression using PaXEvaluation but now I want to take the q^2=0 limit in order to eliminate the UV divergence. Setting explicitly q^2=0 gives me error because of the 1/0 infinite term so I tried to take the limit of the vacuum polarization amplitude for q^2->0 and othin happens. How can I solve the problem? I attach to this message my code for a better comprehension of the problem.
Thanks
![Vacuum_Polarization](https://user-images.githubusercontent.com/81477799/116902585-981cbc00-ac3b-11eb-9100-eeea67778acd.JPG)

**Comment by vsht** [comment]
You can to use the PaXSeries option of PaXEvaluate to expand around q^2=0 when evaluating the integral, cf.
the QED on-shell renormalization example:

https://github.com/FeynCalc/feynhelpers/blob/adcb876fc531cc5b06fe6d0626fe1ac2566d07d4/Examples/QED/QEDRenormalizationOneLoop.m#L250

**Reply by marti2216** [reply]
Thank you very much. If instead I have to compute the remormalized vacuum polarization amplitude in the limit q^2 to infinity how can I do? Because using PaXSeries to expand around infinity returns me an error.

**Reply by vsht** [reply]
Why? At least formally this works
```
(* d/dq^2(q^2 Pi(q^2)) at q^2=Infinity*)
PhotonPi = (8*alphafs*
    Pi^3*(-2*(-2 + D)*A0[me^2] + (4*me^2 + (-2 + D)*pp)*
       B0[pp, me^2, me^2]))/((-1 + D)*pp)

1/((pp)) ((PaXEvaluateUVIRSplit[pp PhotonPi, 
       PaXImplicitPrefactor -> 1/(2 Pi)^D,
       PaXSeries -> {{pp, Infinity, 1}}, PaXAnalytic -> True] - 
      PaXEvaluateUVIRSplit[pp  PhotonPi, 
       PaXImplicitPrefactor -> 1/(2 Pi)^D,
       PaXSeries -> {{pp, Infinity, 0}}, PaXAnalytic -> True]) // 
    FCHideEpsilon) // SimplifyPolyLog
```
Cross-checking this result is of course another story.

**Reply by marti2216** [reply]
Thank you very much. I have just one more question about the vertex renormalization without the division into form factors. I have defined the gamma of the vertex with the options ToPaVe and UsePaVeBasis. Then I set q=0. If I do not use the option UsePaveBasis, when I set q=0 I get the 1/0 error, why? Moreover once I have computed the amplitude the result contains both B0,C0 and C1 even if I apply PaVeReduce. How can I obtain the result in terms of B0 and C0 only? I attach the code for a better understanding of the problem. Thanks
[Vertex.pdf](https://github.com/FeynCalc/feyncalc/files/6421266/Vertex.pdf)

**Reply by vsht** [reply]
The PaVe reduction to A0,B0,C0,D0 implemented in FeynCalc works only for non-vanishing Gram determinants.
Otherwise the output is written in terms of the coefficient functions such as C1 and alike. Unfortunately, 
PaVeReduce is rather limited in this respect. 

If you "trick" TID into doing the reduction with a generic kinematics but later specialize to a particular 
choice that leads to a vanishing Gram determinant, then all kind of inconsistencies (including 1/0) may
arise. So this is rather something to avoid.

**Reply by vsht** [reply]
What is more, even though the zero Gram determinant reduction is not implemented in TID/PaVeReduce, one
can usually achieve the same result using IBPs. I don't have a proof that this always works, but so far I haven't seen
a case where it wouldn't (although I also don't use this so often). Here is a sample code for your C11 using the latest
dev version and FeynHelpers


```
FCClearScalarProducts[]
SPD[r1] = mm2;
SPD[r2] = 0;
SPD[r1, r2] = 0;

(*x1 and x2 are auxiliary momenta "representing" Lorentz indices *)
SPD[r2, x2] = 0;
SPD[r2, x1] = 0;
SPD[x1, x2] = 0;
SPD[x1, r1] = 1;
SPD[x2, r1] = 1;

num = SFAD[{k, mm2}, {k + r1, 0}, {k + r2, mm2}]
num // ToPaVe[#, k, PaVeOrder -> False, PaVeAutoOrder -> False] &

int = SPD[k, x1] SPD[k, x2] 1/(I Pi^2) num

(*so int with the scalar products defined as above represents C11 \
with the desired kinematica*)
TID[int, k]

(*use IBPs to reduce int to something simpler *)
rhs = 
 FIREBurn[int, {k}, {r1, r2, x1, x2}] // FDS[#, k] & // ToPaVe[#, k] &

(*Check using Package-X*)

TID[int, k] // PaXEvaluate // PowerExpand // Simplify
rhs // PaXEvaluate // PowerExpand // Simplify
```
This way one finds
```
PaVe[1, 1, {mm2, mm2, 0}, {mm2, 0, mm2}, PaVeAutoOrder -> True, PaVeAutoReduce -> True] -> -1/2*A0[mm2]/((3 - D)*mm2^2)
```
Defining the correct scalar products usually requires some tinkering, but one could surely automatize this.

**Comment by marti2216** [comment]
Thank you very much for your patience. Unfortunatelly I have one more problem.I am working on the UV divergences of the vertex. I calculate explicitly the counterterm “delta1” and add it to the gamma of the vertex. My first question is: In the definition of “delta1”, should I explicitly say that the paramenter Epsilon is EpsilonUV or can I leave it general?Then I compute the amplitude using this renormalised gamma. Once I use “PaXEvaluateUV[]” the result is an expression proportional to the photon mass squared  which I add in order to take care of the IR divergences. If I set that the mass of the photon to be small with the function “SmallVariable[]”, should Feyncalc eliminate automatically all the terms proportional to that small variable( except logarithms of couse)?As before, please find attach my code for a better comprehension of the problem. Thank you very much, Sincerely,Martina Inviato da Posta per Windows 10 Da: Vladyslav ShtabovenkoInviato: lunedì 10 maggio 2021 13:53A: FeynCalc/feyncalcCc: marti2216; AuthorOggetto: Re: [FeynCalc/feyncalc] Vacuum Polarization (#115) What is more, even though the zero Gram determinant reduction is not implemented in TID/PaVeReduce, onecan usually achieve the same result using IBPs. I don't have a proof that this always works, but so far I haven't seena case where it wouldn't (although I also don't use this so often). Here is a sample code for your C11 using the latestdev version and FeynHelpersFCClearScalarProducts[]SPD[r1] = mm2;SPD[r2] = 0;SPD[r1, r2] = 0; (*x1 and x2 are auxiliary momenta "representing" Lorentz indices *)SPD[r2, x2] = 0;SPD[r2, x1] = 0;SPD[x1, x2] = 0;SPD[x1, r1] = 1;SPD[x2, r1] = 1; num = SFAD[{k, mm2}, {k + r1, 0}, {k + r2, mm2}]num // ToPaVe[#, k, PaVeOrder -> False, PaVeAutoOrder -> False] & int = SPD[k, x1] SPD[k, x2] 1/(I Pi^2) num (*so int with the scalar products defined as above represents C11 \with the desired kinematica*)TID[int, k] (*use IBPs to reduce int to something simpler *)rhs =  FIREBurn[int, {k}, {r1, r2, x1, x2}] // FDS[#, k] & // ToPaVe[#, k] & (*Check using Package-X*) TID[int, k] // PaXEvaluate // PowerExpand // Simplifyrhs // PaXEvaluate // PowerExpand // SimplifyThis way one findsPaVe[1, 1, {mm2, mm2, 0}, {mm2, 0, mm2}, PaVeAutoOrder -> True, PaVeAutoReduce -> True] -> -1/2*A0[mm2]/((3 - D)*mm2^2)Defining the correct scalar products usually requires some tinkering, but one could surely automatize this.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.  
	
      
		Mail priva di virus. www.avast.com

**Reply by vsht** [reply]
From your description it is not entirely clear to me what the problems are, so an explicit example would be much more useful.

It also depends a lot on the scheme you want to use. For 1-loop MSbar renormalization Package-X is not needed,
PaVeUVPart and $KeepLogDivergentScalelessIntegrals are fully sufficient

https://feyncalc.github.io/FeynCalcExamplesMD/Phi4/OneLoop/Renormalization
https://feyncalc.github.io/FeynCalcExamplesMD/Phi3/OneLoop/Renormalization
https://feyncalc.github.io/FeynCalcExamplesMD/QED/OneLoop/Renormalization
https://feyncalc.github.io/FeynCalcExamplesMD/QCD/OneLoop/RenormalizationMassless

SmallVariable is not something very well tested, so again I'd need to see an explicit example
to understand if the behavior is abnormal or not. 

I don't see any attachment here. If you create a zip file of your notebook, you can upload it here directly
via drag and drop.

**Reply by marti2216** [reply]
Sorry, I thought I have attached the code program in the privious message. Here it is 
[Vertex.pdf](https://github.com/FeynCalc/feyncalc/files/6460652/Vertex.pdf)

**Reply by vsht** [reply]
> My first question is: In the definition of “delta1”, should I explicitly say that the paramenter Epsilon is EpsilonUV or can I leave it general?

Epsilon and EpsilonUV are essentially placeholders without extra meaning or properties. PaXEvaluateUV returns only
the UV-divergent piece, hence for convenience the output is given in terms of EpsilonUV. BTW, PaVeUVPart does the
same but without explicit epsilons (it writes D-4 instead). The ordinarty PaXEvaluate returns Epsilon and 
PaXEvaluateUVIRSplit returns both EpsilonUV and EpsilonIR. So it is more the question of how you organize
your calculation and if you care to distinguish between UV and IR singularities or not. For Mathematica/FeynCalc
Epsilon, EpsilolUV and EpsilonIR are just symbols. Cf. `?? Epsilon`, `?? EpsilonUV`, ?? EpsilonIR`

>  If I set that the mass of the photon to be small with the function “SmallVariable[]”, should Feyncalc eliminate automatically all the terms proportional to that small variable( except logarithms of couse)?As before, please find attach my code for a better comprehension of the problem. 

No, unfortunately there is no much automation w.r.t to SmallVariable[]. Essentially, the only function that 
can currently do something with that automatically is `PaVeReduce` which will remove SmallVariables in the numerators.
Cf.

```
PaVe[2, {Pair[Momentum[p, D], Momentum[p, D]], 
  Pair[Momentum[p, D], Momentum[p, D]] + 
   2 Pair[Momentum[p, D], Momentum[q, D]] + 
   Pair[Momentum[q, D], Momentum[q, D]], 
  Pair[Momentum[q, D], Momentum[q, D]]}, {m^2, 
  SmallVariable[\[Lambda]]^2, m^2}, PaVeAutoOrder -> True, 
 PaVeAutoReduce -> True]
% // PaVeReduce
```
vs.
```
PaVe[2, {Pair[Momentum[p, D], Momentum[p, D]], 
  Pair[Momentum[p, D], Momentum[p, D]] + 
   2 Pair[Momentum[p, D], Momentum[q, D]] + 
   Pair[Momentum[q, D], Momentum[q, D]], 
  Pair[Momentum[q, D], Momentum[q, D]]}, {m^2, \[Lambda]^2, m^2}, 
 PaVeAutoOrder -> True, PaVeAutoReduce -> True]
% // PaVeReduce
```
To make use of that you should do something like
```
exp = TID[
  GAD[\[Rho]] . (GSD[p + q + k] + m) . 
    GAD[\[Mu]] . (GSD[p + k] + m) . GAD[\[Rho]]*
   FAD[{k, SmallVariable[\[Lambda]]}, {p + q + k, m}, {p + k, m}], k, 
  UsePaVeBasis -> True, ToPaVe -> True]
PaVeReduce[exp]
```

But I have to admit that I never use SmallVariable myself, so I it is not so much well tested.

You see, for stuff that I don't use in my daily calculations I have to rely on users reporting
bugs. But since most people do not want to invest time into writing bug reports or test fixes,
the situation is as it is.
